File: challenges/2026-09-06-marquee-run-compressor/solution.java
import java.util.*;

public class solution {

    public static String compressMarquee(String s) {
        // TODO: implement
        return "";
    }

    public static void main(String[] args) {
        List<Object[]> tests = new ArrayList<>();
        tests.add(new Object[]{"example 1", "AAAABBBCCD", "4A3BCCD"});
        tests.add(new Object[]{"example 2", "ABCD", "ABCD"});
        tests.add(new Object[]{"example 3", "AABBBBCCCCCC", "AA4B6C"});
        tests.add(new Object[]{"edge: empty string", "", ""});
        tests.add(new Object[]{"edge: long run", "A".repeat(12), "12A"});

        for (Object[] test : tests) {
            String name = (String) test[0];
            String input = (String) test[1];
            String expected = (String) test[2];
            String got = compressMarquee(input);
            String status = got.equals(expected) ? "PASS" : "FAIL";
            System.out.println("[" + status + "] " + name + ": expected=" + expected + " actual=" + got);
        }
    }
}

File: challenges/2026-09-06-marquee-run-compressor/solution.go
package main

import (
	"fmt"
	"strings"
)

func compressMarquee(s string) string {
	// TODO: implement
	return ""
}

type testCase struct {
	name     string
	input    string
	expected string
}

func main() {
	tests := []testCase{
		{"example 1", "AAAABBBCCD", "4A3BCCD"},
		{"example 2", "ABCD", "ABCD"},
		{"example 3", "AABBBBCCCCCC", "AA4B6C"},
		{"edge: empty string", "", ""},
		{"edge: long run", strings.Repeat("A", 12), "12A"},
	}

	for _, tc := range tests {
		got := compressMarquee(tc.input)
		status := "FAIL"
		if got == tc.expected {
			status = "PASS"
		}
		fmt.Printf("[%s] %s: expected=%s actual=%s\n", status, tc.name, tc.expected, got)
	}
}
